src/main/java/io/github/minecraftcursedlegacy/impl/data/DataStorage.java

```diff
@@ -0,0 +1,13 @@
+package io.github.minecraftcursedlegacy.impl.data;
+
+import java.util.function.Supplier;
+
+import io.github.minecraftcursedlegacy.api.data.ModdedData;
+import io.github.minecraftcursedlegacy.api.registry.Id;
+import net.minecraft.util.io.CompoundTag;
+
+public interface DataStorage {
```

Comment:
why's this `Data` but everything else `ModdedData`? (honestly I prefer `Data` over `ModdedData`)